tests/test_experiment_server.py

```diff
@@ -1819,3 +1819,147 @@ def test_does_not_set_participant_end_time_for_participant_otherwise(self, runne
+            [("Home", "dashboard.index"), ("Second", "dashboard.second")]
+        )
+        as_list = list(dashboard_tabs)
+        assert as_list[0] == ("Home", "dashboard.index")
+        assert as_list[1] == ("Second", "dashboard.second")
+
+    def test_dashboard_insert(self, dashboard_tabs):
+        dashboard_tabs.insert("Next", "next")
+        as_list = list(dashboard_tabs)
+        assert as_list[0] == ("Home", "dashboard.index")
+        assert as_list[1] == ("Next", "dashboard.next")
+        dashboard_tabs.insert("Previous", "dashboard.previous", 1)
+        as_list = list(dashboard_tabs)
+        assert as_list[0] == ("Home", "dashboard.index")
+        assert as_list[1] == ("Previous", "dashboard.previous")
+        assert as_list[2] == ("Next", "dashboard.next")
+
+    def test_dashboard_insert_before(self, dashboard_tabs):
+        dashboard_tabs.insert_before_route("First", "first", "index")
+        assert list(dashboard_tabs) == [
+            ("First", "dashboard.first"),
+            ("Home", "dashboard.index"),
+        ]
+
+        dashboard_tabs.insert_before_route(
+            "Second", "dashboard.second", "dashboard.index"
+        )
+        assert list(dashboard_tabs) == [
+            ("Second", "dashboard.second"),
+            ("First", "dashboard.first"),
+            ("Home", "dashboard.index"),
+        ]
+
+    def test_dashboard_insert_after(self, dashboard_tabs):
+        dashboard_tabs.insert_after_route("Last", "last", "index")
+        as_list = list(dashboard_tabs)
+        assert as_list[0] == ("Home", "dashboard.index")
+        assert as_list[1] == ("Last", "dashboard.last")
+        dashboard_tabs.insert_after_route(
+            "Second", "dashboard.second", "dashboard.index"
+        )
+        as_list = list(dashboard_tabs)
+        assert as_list[0] == ("Home", "dashboard.index")
+        assert as_list[1] == ("Second", "dashboard.second")
+        assert as_list[2] == ("Last", "dashboard.last")
+
+    def test_dashboard_remove(self, dashboard_tabs):
+        dashboard_tabs.insert("Last", "last")
+        assert len(list(dashboard_tabs)) == 2
+
+        dashboard_tabs.remove("last")
+        as_list = list(dashboard_tabs)
+        assert len(as_list) == 1
+        assert as_list[0] == ("Home", "dashboard.index")
+
+        dashboard_tabs.insert("Last", "last")
+        assert len(list(dashboard_tabs)) == 2
+
+        dashboard_tabs.remove("dashboard.last")
+        as_list = list(dashboard_tabs)
+        assert len(as_list) == 1
+        assert as_list[0] == ("Home", "dashboard.index")
+
+        dashboard_tabs.remove("index")
+        assert len(list(dashboard_tabs)) == 0
+
+
+class TestDashboard(object):
+    def test_load_user(self):
+        from dallinger.experiment_server.dashboard import admin_user, load_user
+
+        assert admin_user.id == "admin"
+        assert load_user("admin") is admin_user
+        assert load_user("user") is None
+
+    def load_user_from_request(self):
```

Comment:
I'm guessing the omission of `test_` at the beginning of the name is deliberate to prevent it from running for now, but just wanted to make sure.

Reply:
This is updated and fixed, and there are now a bunch of new tests and fixtures for the dashboard.